static/src/javascripts/projects/common/modules/commercial/membership-engagement-banner.js

```diff
@@ -11,19 +11,42 @@ import { isBlocked } from 'common/modules/commercial/membership-engagement-banne
 import { getSync as getGeoLocation } from 'lib/geolocation';
 import { shouldShowReaderRevenue } from 'common/modules/commercial/contributions-utilities';
 import type { Banner } from 'common/modules/ui/bannerPicker';
+import bean from 'bean';
+import fetchJson from 'lib/fetch-json';
 
 import {
     submitComponentEvent,
     addTrackingCodesToUrl,
 } from 'common/modules/commercial/acquisitions-ophan';
 import { acquisitionsBannerControlTemplate } from 'common/modules/commercial/templates/acquisitions-banner-control';
+import userPrefs from 'common/modules/user-prefs';
 
-// change messageCode to force redisplay of the message to users who already closed it.
-const messageCode = 'engagement-banner-2018-07-13';
+type BannerDeployLog = {
+    time: string,
+};
+
+const messageCode = 'engagement-banner';
 
 const canDisplayMembershipEngagementBanner = (): Promise<boolean> =>
     Promise.resolve(shouldShowReaderRevenue());
 
+const getTimestampOfLastBannerDeploy = (): Promise<string> =>
+    fetchJson('/reader-revenue/contributions-banner-deploy-log', {
+        mode: 'cors',
+    }).then((resp: BannerDeployLog) => resp && resp.time);
+
+const hasBannerBeenRedeployedSinceClosed = (): Promise<boolean> =>
+    getTimestampOfLastBannerDeploy()
+        .then(timestamp => {
+            const bannerLastDeployedAt = new Date(timestamp);
+            const userLastClosedBannerAt = new Date(
+                userPrefs.get('engagementBannerLastClosedAt')
+            );
+
+            return bannerLastDeployedAt > userLastClosedBannerAt;
+        })
+        .catch(() => false);
```

Comment:
We might want to consider some logging/alerting in this case, since if something goes wrong then our banner impressions will drop which will cost us money. Perhaps logging to Sentry and then setting up some kind of email alert in Sentry?

Reply:
done!